In [1]:
import pandas as pd
import json
from pprint import pprint

In [2]:
redirects = pd.read_csv("420_redirects.csv", index_col=0)
redirects.head(3)

redir_id  rule_id  start_date                            query  \
0     10303    23358  2020-08-19                   dior хайлайтер   
1     10304    23359  2020-08-19                    dior хайлатер   
2     10319    23374  2020-08-19  diorskin mineral nude luminizer   

   category_id  parent_id    category  
0      3327621  3327595.0  Хайлайтеры  
1      3327621  3327595.0  Хайлайтеры  
2      3327621  3327595.0  Хайлайтеры

In [3]:
#!XL
with open ('sessions_420.json') as f:
    session = json.load(f)

/kernel/lib/python3.7/site-packages/ml_kernel/kernel.py:619: UserWarning: The following variables cannot be serialized: f
  warnings.warn(message)


In [4]:
#!XL
session = pd.read_json('sessions_420.json', orient='records')

In [5]:
session.head(3)

query                                     search_results views  \
0          гель алое      [99730300002, None, 99720100024, 99730300080]    []   
1  бьюти бокс flacon  [None, None, None, None, 19760303130, None, 19...    []   
2           колготки  [22082900011, 22082900010, 22082900009, 220829...    []   

  purchases  
0        []  
1        []  
2        []

In [6]:
query_from_redirects_in_session = session[session['query'].isin(redirects['query'])].reset_index(drop=True)
query_from_redirects_in_session = query_from_redirects_in_session[['query', 'search_results']]
query_from_redirects_in_session['query'].nunique()

244

In [7]:
query_from_redirects_in_session.head(3)

query                                     search_results
0  туалетная вода dior  [7132700003, 7191500004, 7193800001, 197603028...
1            тушь dior  [14104000002, 14104000001, 14104000003, 140540...
2             zelensky  [90151200002, None, None, None, None, 90151200...

В таблице с сессиями 'sessions_420.json' 244 запроса, которые встречаются в таблице с положительными редиректами.

In [8]:
query_from_redirects_in_session.to_csv('query_from_redirects_in_session.csv')

In [9]:
with open("420_searches.json") as ff:
    data = json.load(ff)

/kernel/lib/python3.7/site-packages/ml_kernel/kernel.py:619: UserWarning: The following variables cannot be serialized: ff
  warnings.warn(message)


In [10]:
query = []
product_external_id = []
for i in range(len(data)):
    query.append(list(data[i]['products'].keys()))
    product_external_id.append(data[i]['query'])
    i+=1

In [11]:
searches = pd.DataFrame({'query': product_external_id, 'external_id': query})
searches.head(3)

query                                        external_id
0                  0 70  [89072600018, 89072600022, 89072600015, 890726...
1           2 molecules                                      [26500100001]
2  acqua di gio profumo  [84141400001, 7383500001, 84141400002, 7380100...

In [12]:
query_from_redirects_in_searches = searches[searches['query'].isin(redirects['query'])].reset_index()
query_from_redirects_in_searches.head(3)

index                                       query  \
0     73  dior корректор diorskin forever undercover   
1   1937                                    zelensky   
2   2583                                     лореаль   

                                         external_id  
0  [14105700013, 14105700007, 14105700012, 141057...  
1            [90150300002, 90151300007, 19760301701]  
2                         [19760300042, 24100100033]

In [13]:
query_from_redirects_in_searches = query_from_redirects_in_searches[['query', 'external_id']].explode('external_id').reset_index()
query_from_redirects_in_searches = query_from_redirects_in_searches[['query', 'external_id']]
query_from_redirects_in_searches.head(3)

query  external_id
0  dior корректор diorskin forever undercover  14105700013
1  dior корректор diorskin forever undercover  14105700007
2  dior корректор diorskin forever undercover  14105700012

In [14]:
query_from_redirects_in_searches['query'].nunique()

196

В таблице с сессиями '420_searches.json'  так же два одинаковых запроса, которые встречаются в таблице с положительными редиректами.

In [16]:
query_from_redirects_in_searches.to_csv('query_from_redirects_in_session.csv')